org.hl7.fhir.convertors/src/main/java/org/hl7/fhir/convertors/misc/ProfileVersionAdaptor.java

```diff
@@ -459,7 +458,11 @@ private String adaptPath(String path) {
     if (sd == null) {
       StructureDefinition ssd = sCtxt.fetchTypeDefinition(base);
       if (ssd != null && ssd.getKind() == StructureDefinitionKind.RESOURCE) {
-        return "Basic";
+        if ("CanonicalResource".equals(base)) {
```

Comment:
What is adaptPath actually doing? It looks like it's reducing the path to it's base, but making an exception for Resource. This gets returned to `np` when this is called, but what does `np` mean? `newPath` is my guess, but this doesn't tell me much.

Reply:
this is just adding a special case for a particular kind of past brokenness. I'll rewrite it more clearly in the future when additional infrastructure becomes available